headerCheck/rawHeaderHandler.go

```diff
@@ -65,87 +66,46 @@ func (rh *rawHeaderHandler) GetShardBlockByHash(ctx context.Context, shardId uin
 	return blockHeader, nil
 }
 
-// GetValidatorsInfoPerEpoch will return validators info based on start of
-// epoch metablock for a specific epoch
-func (rh *rawHeaderHandler) GetValidatorsInfoPerEpoch(ctx context.Context, epoch uint32) ([]*state.ShardValidatorInfo, []byte, error) {
-	lastStartOfEpochMetaBlock, err := rh.getLastStartOfEpochMetaBlock(ctx)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	metaBlock, randomness, err := rh.getMetaBlockAndRandomnessForEpoch(ctx, epoch, lastStartOfEpochMetaBlock)
+// GetStartOfEpochMetaBlock will return the start of epoch metablock based on
+// the raw marshalized data from proxy
+func (rh *rawHeaderHandler) GetStartOfEpochMetaBlock(ctx context.Context, epoch uint32) (data.MetaHeaderHandler, error) {
+	metaBlockBytes, err := rh.proxy.GetRawStartOfEpochMetaBlock(ctx, epoch)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	validatorsInfoPerEpoch, err := rh.getValidatorsInfo(ctx, metaBlock)
+	blockHeader := &block.MetaBlock{}
```

Comment:
Valid also GetShardBlockByHash:
we should have the unmarshal mechanism here: first, try the v2 version, if it worked, return it, otherwise, trye with the v1 version of the block.

Can be done in a separate PR.

Reply:
:+1: